analyzers/build/build.go

```diff
@@ -0,0 +1,209 @@
+package build
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"sort"
+
+	"github.com/BurntSushi/toml"
+	"github.com/microcosm-cc/bluemonday"
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/extension"
+)
+
+// IssueMeta represents the issue present in a TOML file.
+type IssueMeta struct {
+	IssueCode   string `toml:"issue_code"`
+	Category    string `toml:"category"`
+	Title       string `toml:"title"`
+	Description string `toml:"description"`
+}
+
+type IssuesMeta []IssueMeta
+
+// IssueTOML is used for decoding issues from a TOML file.
+type IssueTOML struct {
+	Issues []map[string]interface{}
+}
+
+// GenerateTOML helps in generating TOML files for each issue from a TOML file.
+func GenerateTOML() error {
+	// root directory for the repository
+	repoRoot := os.Getenv("REPO_ROOT")
+
+	filename := path.Join(repoRoot, ".deepsource/analyzers/issues.toml")
+	f, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+
+	// fetch issues
+	issues, err := FetchIssues(f)
+	if err != nil {
+		return err
+	}
+
+	// generate TOML files
+	rootDir := path.Join(repoRoot, ".deepsource/analyzers/issues")
+	err = BuildTOML(issues, rootDir)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// FetchIssues reads a TOML file containing all issues, and returns all issues as IssuesMeta.
+func FetchIssues(r io.Reader) (IssuesMeta, error) {
+	issues, err := readTOML(r)
+	if err != nil {
+		return nil, err
+	}
+
+	parsedIssues, err := parseIssues(issues)
+	if err != nil {
+		return nil, err
+	}
+
+	// sort issues (based on issue code) before returning
+	sort.Slice(parsedIssues, func(i, j int) bool {
+		return parsedIssues[i].IssueCode < parsedIssues[j].IssueCode
+	})
+
+	return parsedIssues, nil
+}
+
+// BuildTOML uses issues to generate TOML files to a directory.
+func BuildTOML(issues IssuesMeta, rootDir string) error {
+	if len(issues) == 0 {
+		return errors.New("no issues found")
+	}
+
+	for _, issue := range issues {
+		// The unique identifier (filename) is based on the issue code. TOML files cannot be generated for issues having an invalid/empty code.
+		if issue.IssueCode == "" {
+			return errors.New("invalid issue code. cannot generate toml")
+		}
+
+		// generate file path based on root directory and filename
+		filename := fmt.Sprintf("%s.toml", issue.IssueCode)
+		tomlPath := path.Join(rootDir, filename)
+
+		f, err := os.Create(tomlPath)
+		if err != nil {
+			return err
+		}
+
+		// write to file
+		err = writeTOML(f, issue)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
1. The function name does not make sense.
2. This ideally should have been a method on IssueMetas
3. IssueMeta should have a Write() method that accepts a writer.  writeTOML as a utility function is not the right way to go about this.  
4. The method on `IssueMetas` should range over the issues and call the write method on the IssueMeta object.  

Always make sure you delegate responsibilities to the structs the action is taken on.